tensorcircuit/templates/ensemble.py

```diff
@@ -0,0 +1,165 @@
+"""
+Useful utilities for ensemble
+"""
+
+import tensorflow as tf
+import numpy as np
+
+
+class bagging:  # A.K.A. voting
+    def __init__(self):
+        self.models = []
+        self.is_trained = []
+        self.count = 0
+        self.need_confidence = True  # Help in reducing numbers of get_confidence runs
+        self.permit_train = False
+
+    def append(self, model, is_trained):
+        """
+        Add model to the voting method
+        """
+        self.models.append(model)
+        self.is_trained.append(is_trained)
+        self.count += 1
+
+    def __train_model(self, i, **kwargs):
+        """
+        Train a model if it isn't trained already
+        """
+        if not self.is_trained[i]:
+            self.need_confidence = True
+            self.is_trained[i] = True
+            self.models[i]
+            self.models[i].trainable
+            self.models[i].fit(**kwargs)
+
+    def train(self, **kwargs):
+        """
+        Train all models in the class, **kwargs expect to receive the argus that can be directly sent to tf.fit
+        Expected to be run after finishing compile
+        """
+        if not self.permit_train:
+            raise Exception("Needed to be compiled before training")
+        for i in range(self.count):
+            if "verbose" in kwargs:
```

Comment:
dont train the model labbel is_trained?

Reply:
Yes, the model would not be trained if they are labeled as is_trained true.

Reply:
> Yes, the model would not be trained if they are labeled as is_trained true.

sorry, my bad, not saw the if check in _train function